Refactor dom updates

ErinUntermeyer opened the pull request:
###### What's this PR do?
- The goal of this PR was to refactor any dom updating functions in index.js to follow SRP and DRY principles
- I broke up the checkAvailability function into getDateSelected and checkAvailability
- I broke up filterRoomOnClick into multiple helper functions:
-- getAllAvailableRooms - this one is called when the user clicks 'Check Availability'
-- getFilteredRooms - this one is called whenever a filter button is clicked.
-- getRoomTypeClicked - this one is called from getFilteredRooms
###### Where should the reviewer start?
Most changes were done in index.js
###### How should this be manually tested?
Clone down, run npm install, npm start and go to the url provided
###### Any background context you want to provide?
- My next step is to add the functionality for a customer to reserve a room
- Please let me know if you find any bugs or have any ideas for a better layout for the user. :)
- I know the rooms are currently not sorted upon load. It's on my list!
- If you have a magical CSS rule or resource I could review to fix the background from repeating like that, I'd love to hear it!
###### What are the relevant tickets?
Closes #31 Makes progress on #35

rwilliams659 commented:
A few notes I had as I was testing the app. 

- For the customer dashboard, when I select a date where no rooms are available (1/28/2020 is what I've been using for my testing), no apology message is displayed, it just doesn't display any rooms. 
-For the manager dash, when you search for a user, it looks like what you type must match their name exactly (first and last, case-sensitive). I'm guessing your code tests if the search term === the name, maybe look into includes instead so it doesn't need to be the full name. Also, per your intelligent suggestion to me, if you can also make it not have to be case-sensitive that would be great! 
- As you pointed out, the background does seem to be repeating on both dashboards. Unfortunately, I'm not sure how to fix this :( 
- Otherwise, things are looking awesome. Keep it up!

ErinUntermeyer commented:
> A few notes I had as I was testing the app.
> 
> * For the customer dashboard, when I select a date where no rooms are available (1/28/2020 is what I've been using for my testing), no apology message is displayed, it just doesn't display any rooms.
> * For the manager dash, when you search for a user, it looks like what you type must match their name exactly (first and last, case-sensitive). I'm guessing your code tests if the search term === the name, maybe look into includes instead so it doesn't need to be the full name. Also, per your intelligent suggestion to me, if you can also make it not have to be case-sensitive that would be great!
> * As you pointed out, the background does seem to be repeating on both dashboards. Unfortunately, I'm not sure how to fix this :(
> * Otherwise, things are looking awesome. Keep it up!

* Thanks for the tip on that date that no rooms are available!!! I have added to display the apology to my project board.
* Great ideas for the search functionality, added to the project board as well!
